flowdb/bin/build/0010_schema_public.sql

```diff
@@ -0,0 +1,147 @@
+/*
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+/*
+PUBLIC ---------------------------------------------
+
+Here we define the public schema. This schema
+
+contains:
+
+- files
+ this table is designed for the purposes of keeping track
+    of the ingestion of text files.
+
+    Available columns:
+
+      - files:  where information about the ingestion of
+                files is kept.
+
+- d_date:                   General purpose date dimension table.
+- d_time:                   General purpose time dimension table.
+
+----------------------------------------------------
+*/
+CREATE TABLE IF NOT EXISTS public.files(
+    
+    id TEXT PRIMARY KEY,
+    full_path TEXT,
+    directory_path TEXT,
+    specification_path TEXT,
+    target_table TEXT,
+    created TIMESTAMPTZ,
+    processed BOOLEAN,
+    datetime TIMESTAMPTZ,
+    size NUMERIC,
+    status TEXT,
+    record_number NUMERIC,
+    log TEXT
+
+    );
+
+    CREATE TABLE d_date
+    (
+      date_dim_id              INT PRIMARY KEY,
+      date_actual              DATE NOT NULL,
+      epoch                    BIGINT NOT NULL,
+      day_suffix               VARCHAR(4) NOT NULL,
+      day_name                 VARCHAR(9) NOT NULL,
+      day_of_week              INT NOT NULL,
+      day_of_month             INT NOT NULL,
+      day_of_quarter           INT NOT NULL,
+      day_of_year              INT NOT NULL,
+      week_of_month            INT NOT NULL,
+      week_of_year             INT NOT NULL,
+      week_of_year_iso         CHAR(10) NOT NULL,
+      month_actual             INT NOT NULL,
+      month_name               VARCHAR(9) NOT NULL,
+      month_name_abbreviated   CHAR(3) NOT NULL,
+      quarter_actual           INT NOT NULL,
+      quarter_name             VARCHAR(9) NOT NULL,
+      year_actual              INT NOT NULL,
+      first_day_of_week        DATE NOT NULL,
```

Comment:
Weeks are Monday to Sunday? Would we adjust for places where 'weekends' are defined differently because e.g. Friday is a holy day? (I don't actually have a view about this at the moment, just ruminating for now...)

Reply:
Well I guess past a certain point it is on the deployer to modify the dimension to suit their needs - even at the several decades size populated here, it isn't a big deal to update values or even add a new column. Weeks are populated as ISO standard, so Monday is first day.